src/java/picard/illumina/ClusterDataToSamConverter.java

```diff
@@ -134,12 +169,29 @@ private SAMRecord createSamRecord(final ReadData readData, final String readName
             sam.setAttribute(SAMTag.BC.name(), unmatchedBarcode);
         }
 
-        if (molecularIndex != null) {
+        if (!molecularIndexes.isEmpty()) {
             //TODO: add RX and QX to the list of SAMTags and change this. initial discussion
             //TODO: here: https://github.com/broadinstitute/picard/issues/287
             //TODO: HTS-spec issue: https://github.com/samtools/hts-specs/issues/109
-            sam.setAttribute("RX", molecularIndex);
-            sam.setAttribute("QX", molecularIndexQ);
+            sam.setAttribute(this.molecularIndexTag, String.join("", molecularIndexes));
+            sam.setAttribute(this.molecularIndexQualityTag, String.join("", molecularIndexQualities));
+            if (!this.tagPerMolecularIndex.isEmpty() && molecularIndexes.size() > 1) {
```

Comment:
I suppose I don't really need it, but the thinking was this would `act` the same as `molecularIndexTag`.  I'll remove it.